app/controllers/registrations_controller.rb

```diff
@@ -576,52 +576,59 @@ def stripe_webhook
+    payment_account = competition.payment_account_for(payment_integration.to_sym)
 
-    unless stored_record.present?
-      flash[:error] = t("registrations.payment_form.errors.stripe_not_found")
-      return redirect_to competition_register_path(competition_id)
+    unless payment_account.present?
+      flash[:danger] = t("registrations.payment_form.errors.cpi_disconnected")
+      return redirect_to competition_register_path(competition)
     end
 
-    unless stored_record.payment_intent?
-      flash[:error] = t("registrations.payment_form.errors.stripe_not_an_intent")
-      return redirect_to competition_register_path(competition_id)
+    begin
+      stored_record = payment_account.find_payment_from_request(params)
+    rescue WcaExceptions::PaymentInvalidError => e
+      flash[:error] = e.message
+      return redirect_to competition_register_path(competition)
     end
 
-    stored_intent = stored_record.payment_intent
+    unless stored_record.present?
+      flash[:error] = t("registrations.payment_form.errors.generic.not_found", provider: t("payments.payment_providers.#{payment_integration}"))
+      return redirect_to competition_register_path(competition)
+    end
 
-    registration = stored_intent.holder
-    uses_v2 = registration.is_a? MicroserviceRegistration
+    stored_intent = stored_record.payment_intent
 
-    unless stored_intent.client_secret == intent_secret
-      flash[:error] = t("registrations.payment_form.errors.stripe_secret_invalid")
-      return redirect_to competition_register_path(competition_id)
+    unless stored_intent.present?
+      flash[:error] = t("registrations.payment_form.errors.generic.intent_not_found", provider: t("payments.payment_providers.#{payment_integration}"))
+      return redirect_to competition_register_path(competition)
     end
 
-    # No need to create a new intent here. We can just query the stored intent from Stripe directly.
-    stripe_intent = stored_intent.retrieve_remote
+    remote_intent = stored_intent.retrieve_remote
 
-    unless stripe_intent.present?
-      flash[:error] = t("registrations.payment_form.errors.stripe_not_found")
-      return redirect_to competition_register_path(competition_id)
+    unless remote_intent.present?
+      flash[:error] = t("registrations.payment_form.errors.generic.remote_not_found", provider: t("payments.payment_providers.#{payment_integration}"))
+      return redirect_to competition_register_path(competition)
     end
 
-    stored_intent.update_status_and_charges(stripe_intent, current_user) do |charge_transaction|
+    registration = stored_intent.holder
+
+    stored_intent.update_status_and_charges(remote_intent, current_user) do |charge_transaction|
       ruby_money = charge_transaction.money_amount
 
-      if uses_v2
+      if registration.is_a? MicroserviceRegistration
         begin
           Microservices::Registrations.update_registration_payment(
```

Comment:
Also I think this method is now outdated given that we're using v3 registrations?